packages/blade/src/components/Button/BaseButton/BaseButton.tsx

```diff
@@ -31,6 +36,103 @@ type BaseButtonWithIconProps = BaseButtonCommonProps & {
 
 export type BaseButtonProps = BaseButtonWithIconProps | BaseButtonWithoutIconProps;
 
+const StyledBaseText = styled(BaseText)(
+  ({
+    hasIcon,
+    iconPosition,
+    iconSpacing,
+  }: BaseTextProps &
+    Pick<BaseButtonProps, 'iconPosition'> & { hasIcon: boolean; iconSpacing: string }) => `
+    padding-left: ${hasIcon && iconPosition === 'left' ? iconSpacing : 0};
+    padding-right: ${hasIcon && iconPosition === 'right' ? iconSpacing : 0};
+`,
+);
+
+type ConfigProps = {
+  iconSize: IconSize; // prop
+  fontSize: keyof Theme['typography']['fonts']['size']; // prop
+  lineHeight: keyof Theme['typography']['lineHeights']; // prop
+  buttonHeight: '48px' | '40px' | '32px' | '28px'; // raw value
+  iconSpacing: string; // raw value
+  iconColor: IconProps['color']; // prop
+  textColor: BaseTextProps['color']; // prop
+  buttonSpacing: string; // raw value
+  text?: string;
+};
+
+const getProps = ({
+  theme,
+  size,
+  children,
+  buttonTypographyTokens,
+}: {
+  theme: Theme;
+  size: BaseButtonCommonProps['size'];
+  children?: string;
+  buttonTypographyTokens: typeof typography.onDesktop | typeof typography.onMobile;
```

Comment:
put the typings under tokens file itself and then use it here?

Reply:
it's getting very confusing to context switch and build context as to what is what in which PR 😭

Reply:
We can do everything in a single PR but it would be too huge to review then